spec/controllers/passwords_controller_spec.rb

```diff
@@ -3,186 +3,161 @@
+        email = ActionMailer::Base.deliveries.last
+        expect(email.subject).to match(/change your password/i)
       end
+    end
 
-      describe 'with incorrect email address' do
-        before do
-          email = 'user1@example.com'
-          user = Clearance.configuration.user_model.exists?(email: email)
-          expect(user).not_to be_present
+    context "email does not belong to an existing user" do
+      it "does not deliver an email" do
+        ActionMailer::Base.deliveries.clear
+        email = "this_user_does_not_exist@non_existent_domain.com"
 
-          ActionMailer::Base.deliveries.clear
-          expect(@user.reload.confirmation_token).to eq @user.confirmation_token
+        post :create, password: { email: email }
 
-          post :create, password: { email: email }
-        end
+        expect(ActionMailer::Base.deliveries).to be_empty
+      end
 
-        it 'should not generate a token for the change your password email' do
-          expect(@user.reload.confirmation_token).to eq @user.confirmation_token
-        end
+      it "still responds with success so as not to leak registered users" do
+        email = "this_user_does_not_exist@non_existent_domain.com"
 
-        it 'should not send a password reminder email' do
-          expect(ActionMailer::Base.deliveries).to be_empty
-        end
+        post :create, password: { email: email }
 
-        it { is_expected.to render_template(:create) }
+        expect(response).to be_success
+        expect(response).to render_template "passwords/create"
       end
     end
   end
 
-  describe 'a signed up user and forgotten password' do
-    before do
-      @user = create(:user)
-      @user.forgot_password!
-    end
+  describe "#edit" do
+    context "valid id and token are supplied" do
+      it "renders the password form for the user" do
+        user = create(:user, :with_forgotten_password)
 
-    describe 'on GET to #edit with correct id and token' do
-      before do
-        get :edit,
-            user_id: @user.to_param,
-            token: @user.confirmation_token
-      end
+        get :edit, user_id: user, token: user.confirmation_token
 
-      it 'should find the user' do
-        expect(assigns(:user)).to eq @user
+        expect(response).to be_success
+        expect(response).to render_template(:edit)
+        expect(assigns(:user)).to eq user
       end
-
-      it { is_expected.to respond_with(:success) }
-      it { is_expected.to render_template(:edit) }
     end
 
-    describe 'on GET to #edit with correct id but blank token' do
-      before do
-        get :edit, user_id: @user.to_param, token: ''
-      end
-
-      it { is_expected.to set_the_flash.to(/double check the URL/i).now }
-      it { is_expected.to render_template(:new) }
-    end
+    context "blank token is supplied" do
+      it "renders the new password reset form with a flash notice" do
+        get :edit, user_id: 1, token: ''
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.